src/main/java/org/tikv/txn/gc/GCWorker.java

```diff
@@ -0,0 +1,467 @@
+  }
+
+  public void start() {
+    logger.info(String.format("[gc worker] %s starts", uuid));
+    scheduler.scheduleAtFixedRate(() -> {
+      try {
+        logger.debug("[gc worker] tick");
+        pdClient.keepLeaseAlive(l_uuid);
+        tick();
+      } catch (Exception e) {
+        logger.warn("[gc worker] gc fails to proceed", e);
+      }
+    }, 0, 1, TimeUnit.SECONDS);
+  }
+
+  private void tick() {
+    boolean isLeader = checkLeader();
+    if (isLeader) {
+      try {
+        leaderTick();
+      } catch (Exception e) {
+        logger.warn("[gc worker] check leader err: ", e);
+      }
+    }
+  }
+
+  private void leaderTick() {
+    if (gcIsRunning) {
+      logger.info(String.format("[gc worker] leader tick on %s: there's already a gc job running. skipped.", uuid));
+      return;
+    }
+
+    if (getTimestamp().getPhysical() < lastFinish + gcWaitTime) {
+      gcIsRunning = false;
+      logger.info(String.format("[gc worker] leader tick on %s: another gc job has just finished. skipped.", uuid));
+      return;
+    }
+
+    long safePoint;
+    try {
+      safePoint = prepare();
+    } catch (Exception e) {
+      logger.warn("[gc worker] leader tick err: %s", e);
+      return;
+    }
+    if (safePoint == 0) {
+      gcIsRunning = false;
+      return;
+    }
+
+    gcIsRunning = true;
+    logger.info(String.format("[gc worker] %s starts the whole job, safePoint: %d", uuid, safePoint));
+    runGCJob(safePoint);
+    lastFinish = System.currentTimeMillis();
+    gcIsRunning = false;
+  }
+
+  private boolean checkLeader() {
+    // acquire lock
+    boolean ok = pdClient.lock(GC_LEADER_UUID_KEY, l_uuid) && doCheckLeader();
+    // release lock
+    pdClient.unlock(GC_LEADER_UUID_KEY);
+    return ok;
+  }
+
+  private boolean doCheckLeader() {
+    long leader = getLong(GC_LEADER_UUID_KEY);
+    logger.debug(String.format("[gc worker] got leader: %s", Long.toHexString(leader)));
+    if (leader == l_uuid) {
+      try {
+        putLong(GC_LEADER_LEASE_KEY, System.currentTimeMillis() + gcWorkerLease);
+      } catch (Exception e) {
+        return false;
+      }
+      return true;
+    }
+    long lease = getLong(GC_LEADER_LEASE_KEY);
+    if (lease == 0 || lease < System.currentTimeMillis()) {
+      logger.debug(String.format("[gc worker] register %s as leader", uuid));
+      try {
+        pdClient.txn().Then(
+            toPutOp(GC_LEADER_UUID_KEY, uuid),
+            toPutOp(GC_LEADER_LEASE_KEY, System.currentTimeMillis() + gcWorkerLease)
+        ).commit().get(500, TimeUnit.MILLISECONDS);
+      } catch (Exception e) {
+        return false;
+      }
+      return true;
+    }
+    return false;
+  }
+
+  private void runGCJob(long safePoint) {
+    try {
+      resolveLocks(safePoint);
+    } catch (Exception e) {
+      logger.error(String.format("[gc worker] %s resolve locks returns an error %s", uuid, e));
+      return;
+    }
+    doGC(safePoint);
```

Comment:
From TiKV 3.0, "Distributed GC" has been supported. So for TiKV 3.0 clusters, after resolving locks, you can simply upload the safepoint to PD via PD's "uploadGCSafePoint" API

Reply:
I can make it configurable in another PR.